src/lib.rs

```diff
@@ -192,11 +198,7 @@ impl GpuProfiler {
         let frame = self.pending_frames.first_mut()?;
 
         // We only process if all mappings succeed.
-        if frame
-            .query_pools
-            .iter_mut()
-            .any(|pool| block_on(poll_once(pool.buffer_mapping.as_mut().unwrap())).is_none())
-        {
+        if frame.mapped_buffers.load(std::sync::atomic::Ordering::SeqCst) != frame.query_pools.len() {
```

Comment:
should be Ordering::Acquire, rationale see comment above - here we want to acquire the state that got published earlier